irohad/pending_txs_storage/impl/pending_txs_storage_impl.cpp

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "pending_txs_storage/impl/pending_txs_storage_impl.hpp"
+
+#include "multi_sig_transactions/state/mst_state.hpp"
+
+namespace iroha {
+
+  PendingTransactionStorageImpl::PendingTransactionStorageImpl(
+      StateObservable updatedBatches,
+      BatchObservable preparedBatch,
+      BatchObservable expiredBatch) {
+    updated_batches_subscription_ =
+        updatedBatches.subscribe([this](const SharedState &updatedBatches) {
+          this->updatedBatchesHandler(updatedBatches);
+        });
+    prepared_batch_subscription_ =
+        preparedBatch.subscribe([this](const SharedBatch &preparedBatch) {
+          this->preparedBatchHandler(preparedBatch);
+        });
+    expired_batch_subscription_ =
+        expiredBatch.subscribe([this](const SharedBatch &expiredBatch) {
+          this->expiredBatchHandler(expiredBatch);
+        });
+  }
+
+  PendingTransactionStorageImpl::~PendingTransactionStorageImpl() {
+    updated_batches_subscription_.unsubscribe();
+    prepared_batch_subscription_.unsubscribe();
+    expired_batch_subscription_.unsubscribe();
+  }
+
+  PendingTransactionStorageImpl::SharedTxsCollectionType
+  PendingTransactionStorageImpl::getPendingTransactions(
+      const AccountIdType &accountId) const {
+    std::shared_lock<std::shared_timed_mutex> lock(mutex_);
+    auto creator_it = storage_.index.find(accountId);
+    if (storage_.index.end() != creator_it) {
+      auto &batch_hashes = creator_it->second;
+      SharedTxsCollectionType result;
+      auto &batches = storage_.batches;
+      for (const auto &batch_hash : batch_hashes) {
+        auto batch_it = batches.find(batch_hash);
+        if (batches.end() != batch_it) {
+          auto &txs = batch_it->second->transactions();
+          result.insert(result.end(), txs.begin(), txs.end());
+        }
+      }
+      return result;
+    }
+    return {};
+  }
+
+  std::set<PendingTransactionStorageImpl::AccountIdType>
+  PendingTransactionStorageImpl::batchCreators(
+      const TransactionBatch &batch) const {
```

Comment:
Why isn't this function static?

Reply:
There are no external users. Let's do it static function.